src/multivariate/mvnormal.jl

```diff
@@ -26,30 +26,30 @@ abstract AbstractMvNormal <: ContinuousMultivariateDistribution
 
 ### Generic methods (for all AbstractMvNormal subtypes)
 
-insupport(d::AbstractMvNormal, x::AbstractVector) =
+insupport{T<:Real}(d::AbstractMvNormal, x::AbstractVector{T}) =
     length(d) == length(x) && allfinite(x)
 
 mode(d::AbstractMvNormal) = mean(d)
 modes(d::AbstractMvNormal) = [mean(d)]
 
-entropy(d::AbstractMvNormal) = (length(d) * (Float64(log2π) + 1) + logdetcov(d))/2
+entropy(d::AbstractMvNormal) = 0.5 * (length(d) * (Float64(log2π) + 1.0) + logdetcov(d))
```

Comment:
Avoid `Float64` literals since it will promote in cases where it is not necessary. You can just divide with `2`. LLVM is smart enough to optimize it to a multiplication when possible, e.g.
```jl


julia> f(x) = x/2
f (generic function with 1 method)

julia> @code_llvm f(2.0)

define double @julia_f_68028(double) #0 {
top:
  %1 = fmul double %0, 5.000000e-01
  ret double %1
}
```
 `log2π` is bit tricky so maybe start out with computing `ldcd = logdetcov(d)` and use `typeof(ldcd)(log2π)`.